llvm/lib/DebugInfo/PDB/Native/GSIStreamBuilder.cpp

```diff
@@ -319,7 +320,13 @@ Error GSIStreamBuilder::finalizeMsfLayout() {
     return Idx.takeError();
   PublicsStreamIndex = *Idx;
 
-  uint32_t RecordBytes = PSH->RecordByteSize + GSH->RecordByteSize;
+  uint64_t RecordBytes = PSH->RecordByteSize + GSH->RecordByteSize;
+  if (RecordBytes > UINT32_MAX)
+    return make_error<StringError>(
+        formatv("the public ({0} bytes) and global ({1} bytes) "
```

Comment:
Can we also display what the maximum size is? It’s obvious for us, but not for regular users.

Reply:
Done.

Reply:
Oops. Done.